server/src/test/java/org/opensearch/index/store/remote/file/OnDemandBlockSnapshotIndexInputTests.java

```diff
@@ -0,0 +1,510 @@
+            blockSizeShift,
+            fileSize
+        );
+        runAllTestsFor(ondemandBlockSnapshotIndexInput_4KB, 1 << blockSizeShift, fileSize);
+        cleanDirectory(path);
+
+        // block size 1MB
+        blockSizeShift = 20;
+        OnDemandBlockSnapshotIndexInput ondemandBlockSnapshotIndexInput_1MB = createOnDemandBlockSnapshotIndexInput(
+            blockSizeShift,
+            fileSize
+        );
+        runAllTestsFor(ondemandBlockSnapshotIndexInput_1MB, 1 << blockSizeShift, fileSize);
+        cleanDirectory(path);
+
+        // block size 4MB
+        blockSizeShift = 22;
+        OnDemandBlockSnapshotIndexInput ondemandBlockSnapshotIndexInput_4MB = createOnDemandBlockSnapshotIndexInput(
+            blockSizeShift,
+            fileSize
+        );
+        runAllTestsFor(ondemandBlockSnapshotIndexInput_4MB, 1 << blockSizeShift, fileSize);
+        cleanDirectory(path);
+    }
+
+    public void runAllTestsFor(OnDemandBlockSnapshotIndexInput blockedSnapshotFile, int blockSize, int fileSize) throws Exception {
+        TestGroup.testGetBlock(blockedSnapshotFile, blockSize, fileSize);
+        TestGroup.testGetBlockOffset(blockedSnapshotFile, blockSize, fileSize);
+        TestGroup.testGetBlockStart(blockedSnapshotFile, blockSize, fileSize);
+        TestGroup.testCurrentBlockStart(blockedSnapshotFile, blockSize, fileSize);
+        TestGroup.testCurrentBlockPosition(blockedSnapshotFile, blockSize, fileSize);
+        TestGroup.testClone(blockedSnapshotFile, blockSize, fileSize);
+        TestGroup.testSlice(blockedSnapshotFile, blockSize, fileSize);
+        TestGroup.testGetFilePointer(blockedSnapshotFile, blockSize, fileSize);
+        TestGroup.testReadByte(blockedSnapshotFile, blockSize, fileSize);
+        TestGroup.testReadShort(blockedSnapshotFile, blockSize, fileSize);
+        TestGroup.testReadInt(blockedSnapshotFile, blockSize, fileSize);
+        TestGroup.testReadLong(blockedSnapshotFile, blockSize, fileSize);
+        TestGroup.testReadVInt(blockedSnapshotFile, blockSize, fileSize);
+        TestGroup.testSeek(blockedSnapshotFile, blockSize, fileSize);
+        TestGroup.testReadByteWithPos(blockedSnapshotFile, blockSize, fileSize);
+        TestGroup.testReadShortWithPos(blockedSnapshotFile, blockSize, fileSize);
+        TestGroup.testReadIntWithPos(blockedSnapshotFile, blockSize, fileSize);
+        TestGroup.testReadLongWithPos(blockedSnapshotFile, blockSize, fileSize);
+        TestGroup.testReadBytes(blockedSnapshotFile, blockSize, fileSize);
+    }
+
+    // create OnDemandBlockSnapshotIndexInput for each block size
+    private static OnDemandBlockSnapshotIndexInput createOnDemandBlockSnapshotIndexInput(int blockSizeShift, long fileSize) {
+
+        // file info should be initialized per test method since file size need to be calculated
+        fileInfo = mock(BlobStoreIndexShardSnapshot.FileInfo.class);
+        byteSizeValue = mock(ByteSizeValue.class);
```

Comment:
These are plain data objects as far as I can tell. Why mock them? It would be better just to construct real instances. Also why are they static fields? I think they can just be local variables here since they are just given to the OnDemandBlockSnapshotIndexInput instance returned here. I think the same is true of TransferManager as well.

Reply:
resolved all except TransferManager as within the scope of this unit test, there is no need to construct TransferManager by passing another mocked blob container and executor service. I think it is better to mock it in this scope

Reply:
Sorry, that wasn't clear. I meant that TransferManager could be a local variable, though still mocked. No need for it to be a mutable static field.